core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1080,6 +1080,21 @@ abstract class RDD[T: ClassTag](
       var partiallyAggregated = mapPartitions(it => Iterator(aggregatePartition(it)))
       var numPartitions = partiallyAggregated.partitions.length
       val scale = math.max(math.ceil(math.pow(numPartitions, 1.0 / depth)).toInt, 2)
+
+      // Do one level of aggregation based on executorId before starting the tree
+      // NOTE: exclude the driver from list of executors
+      val numExecutors = math.max(context.getExecutorStorageStatus.length - 1, 1)
```

Comment:
it might be good to explicitly filter out the driver `SparkContext.DRIVER_IDENTIFIER` so we don't silently have the wrong number if we change this in the future